e2e_playwright/st_audio_test.py

```diff
@@ -61,3 +61,27 @@ def test_audio_end_time_loop(app: Page):
     app.wait_for_timeout(6000)
     expect(audio_element).to_have_js_property("paused", False)
     wait_until(app, lambda: 16 < audio_element.evaluate("el => el.currentTime") < 18)
+
+
+@pytest.mark.skip_browser("webkit")
+def test_audio_autoplay(app: Page, assert_snapshot: ImageCompareFunction):
+    """Test that `st.audio` autoplay property works correctly."""
+    audio_elements = app.get_by_test_id("stAudio")
+    expect(audio_elements).to_have_count(4)
+
+    expect(audio_elements.nth(3)).to_be_visible()
+
+    audio_element = audio_elements.nth(3)
+    expect(audio_element).to_have_js_property("paused", True)
+    expect(audio_element).to_have_js_property("autoplay", False)
+
+    checkbox_element = app.get_by_test_id("stCheckbox")
+    autoplay_checkbox = checkbox_element.nth(0)
+    autoplay_checkbox.click()
+    # To prevent flakiness, we wait for the audio to load and start playing
+    wait_until(app, lambda: audio_element.evaluate("el => el.readyState") == 4)
+    expect(audio_element).to_have_js_property("paused", False)
+    expect(audio_element).to_have_js_property("autoplay", True)
+    # Wait for the audio to play for at least 1 second
+    wait_until(app, lambda: audio_element.evaluate("el => el.currentTime") > 1)
+    assert_snapshot(audio_element, name="st_audio-autoplay")
```

Comment:
The e2e test looks great, but this screenshot might get a bit flaky since I'm not sure if we can rely on it always having the exact same time. Maybe instead we just do another paused == False check to make sure that its still playing and leave out the screenshot:

```
expect(audio_element).to_have_js_property("paused", False)
```